x-pack/plugins/data_visualizer/public/application/data_drift/document_count_chart_single_brush/document_count_chart_singular.tsx

```diff
@@ -0,0 +1,499 @@
+          setOriginalWindowParameters(wpSnap);
+          setWindowParameters(wpSnap);
+
+          if (brushSelectionUpdateHandler !== undefined) {
+            brushSelectionUpdateHandler(wpSnap, true);
+          }
+        }
+      }
+    },
+    [
+      interval,
+      timeRangeEarliest,
+      timeRangeLatest,
+      snapTimestamps,
+      originalWindowParameters,
+      setWindowParameters,
+      brushSelectionUpdateHandler,
+      adjustedChartPoints,
+      timefilterUpdateHandler,
+      viewMode,
+      windowParameters,
+    ]
+  );
+
+  const onElementClick: ElementClickListener = useCallback(
+    ([elementData]) => {
+      if (brushSelectionUpdateHandler === undefined) {
+        return;
+      }
+      const startRange = (elementData as XYChartElementEvent)[0].x;
+
+      triggerAnalysis(startRange);
+    },
+    [triggerAnalysis, brushSelectionUpdateHandler]
+  );
+
+  useEffect(() => {
+    if (autoAnalysisStart !== undefined) {
+      triggerAnalysis(autoAnalysisStart);
+    }
+  }, [triggerAnalysis, autoAnalysisStart]);
+
+  useEffect(() => {
+    if (isBrushCleared && originalWindowParameters !== undefined) {
+      setOriginalWindowParameters(undefined);
+      setWindowParameters(undefined);
+    }
+  }, [isBrushCleared, originalWindowParameters]);
+
+  function onWindowParametersChange(
+    wp: SingleBrushWindowParameters,
+    wpPx: SingleBrushWindowParameters
+  ) {
+    if (brushSelectionUpdateHandler === undefined) {
+      return;
+    }
+    setWindowParameters(wp);
+    brushSelectionUpdateHandler(wp, false);
+  }
+
+  const [mlBrushWidth, setMlBrushWidth] = useState<number>();
+  const [mlBrushMarginLeft, setMlBrushMarginLeft] = useState<number>();
+
+  useEffect(() => {
+    if (viewMode !== VIEW_MODE.BRUSH) {
+      setOriginalWindowParameters(undefined);
+      setWindowParameters(undefined);
+    }
+  }, [viewMode]);
+
+  const isBrushVisible =
+    originalWindowParameters &&
+    windowParameters &&
+    mlBrushMarginLeft &&
+    mlBrushWidth &&
+    mlBrushWidth > 0;
+
+  const barColor = barColorOverride ? [barColorOverride] : undefined;
+  const barHighlightColor = barHighlightColorOverride ? [barHighlightColorOverride] : ['orange'];
+
+  return (
+    <>
+      {isBrushVisible ? (
+        <div className="dataDriftSingleBrush" data-test-subj={'dataDriftSingleBrush'}>
+          <div
+            css={{
+              'margin-bottom': '-4px',
+            }}
+          >
+            <SingleBrush
+              id={id}
+              windowParameters={originalWindowParameters}
+              min={timeRangeEarliest}
+              max={timeRangeLatest + interval}
+              onChange={onWindowParametersChange}
+              marginLeft={mlBrushMarginLeft}
+              snapTimestamps={snapTimestamps}
+              width={mlBrushWidth}
+            />
+          </div>
+        </div>
+      ) : (
+        <EuiText color="subdued" size="s" textAlign="center">
```

Comment:
Should we only allow the analysis to be run when a selection is made in both charts? Currently it looks like a default range is set in the Comparison chart on selection in the Reference chart but the brush only appears when you perform a click in the Reference chart. The other option would be to add brushes to both charts when the first click is made in either chart.

<img width="954" alt="image" src="https://github.com/elastic/kibana/assets/7405507/f8e6113f-b390-4756-8593-aa03982456dc">